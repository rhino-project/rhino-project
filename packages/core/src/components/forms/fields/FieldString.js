import { useGlobalComponent } from 'rhino/hooks/overrides';
import { FieldInputBase } from './FieldInput';

export const FieldStringBase = (props) => (
  <FieldInputBase type="text" {...props} />
);

const FieldString = (props) =>
  useGlobalComponent('FieldString', FieldStringBase, props);

export default FieldString;
