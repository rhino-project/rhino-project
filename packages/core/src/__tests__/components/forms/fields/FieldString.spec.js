import FieldString from '../../../../components/forms/fields/FieldString';
import { sharedFieldTests } from './sharedFieldTests';

describe('FieldString', () => {
  sharedFieldTests(FieldString);
});
