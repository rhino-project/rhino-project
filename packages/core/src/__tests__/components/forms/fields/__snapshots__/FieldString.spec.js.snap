// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FieldString > render with place holder 1`] = `
<DocumentFragment>
  <input
    aria-invalid="false"
    autocomplete="off"
    class="form-control"
    id="dummy"
    name="dummy"
    path="dummy"
    placeholder="placeholder"
    type="text"
  />
</DocumentFragment>
`;

exports[`FieldString > renders inside of model context 1`] = `
<DocumentFragment>
  <input
    aria-invalid="false"
    autocomplete="off"
    class="form-control"
    id="dummy"
    name="dummy"
    path="dummy"
    type="text"
  />
</DocumentFragment>
`;

exports[`FieldString > should render with global override shorthand 1`] = `
<DocumentFragment>
  <div>
    Bar
  </div>
</DocumentFragment>
`;
